homeassistant/components/xbox_live/sensor.py

```diff
@@ -24,65 +28,76 @@
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Xbox platform."""
-    from xboxapi import xbox_api
-
-    api = xbox_api.XboxApi(config.get(CONF_API_KEY))
-    devices = []
+    api = xbox_api.XboxApi(config[CONF_API_KEY])
+    entities = []
 
     # request personal profile to check api connection
     profile = api.get_profile()
     if profile.get("error_code") is not None:
         _LOGGER.error(
             "Can't setup XboxAPI connection. Check your account or "
-            " api key on xboxapi.com. Code: %s Description: %s ",
-            profile.get("error_code", STATE_UNKNOWN),
-            profile.get("error_message", STATE_UNKNOWN),
+            "api key on xboxapi.com. Code: %s Description: %s ",
+            profile.get("error_code", "unknown"),
+            profile.get("error_message", "unknown"),
         )
         return
 
-    for xuid in config.get(CONF_XUID):
-        new_device = XboxSensor(hass, api, xuid)
-        if new_device.success_init:
-            devices.append(new_device)
+    users = config[CONF_XUID]
+
+    interval = timedelta(minutes=1 * len(users))
+    interval = config.get(CONF_SCAN_INTERVAL, interval)
+
+    for xuid in users:
+        gamercard = get_user_gamercard(api, xuid)
+        if gamercard is None:
```

Comment:
Maybe drop a log message at this point? Might be helpful in case the user adds a xuid that cannot be found.

Reply:
Nvm, handled below. 🤦‍♂